src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Undo the most recent command.
+ */
+public class UndoCommand extends Command {
+
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + "Undo the most recent command "
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_UNDO_SUCCESS = "Successfully undo the most recent command";
+    public static final String MESSAGE_UNDO_FAIL = "Cannot undo";
+
```

Comment:
Since the undo command currently only works with every command except for find and list, you may want to consider adding that information into the message usage and message undo fail so that the user will know exactly why it fails :)

Reply:
Ok noted!